3773 add has block functions

andreilupu opened the pull request:
## Description
I know that #3781 is there, but it's a stale PR for these handy functions, and I could not manage to fix that conflicting branch to continue the job from there.

I managed to address some changes requested in the comments there and some PHPUnit tests.
But, I'm not sure that if these functions should have a `_type` suffix since they are block types, not just blocks.( I would vote for `gutenberg_content_has_block_type` ).

Intentionally, I've switched the parameters order because it is more natural to have the required parameters first( the block name) and the optional ones after.

I stick to the idea that these functions should be fast, and if in later versions there will be a need for strict matching we could build functions like `wp_block_match` where we could also match block attributes.

With all the props to @ideadude for inspiration and idea.

## How has this been tested?
It's a PHP feature, so I tested it via PHPUnit.

## Screenshots <!-- if applicable -->

## Types of changes
New feature (a non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

andreilupu committed:
Add gutenberg_content_has_block and gutenberg_post_has_block functionality on server side.

Also, added some phpunit tests for these functions.

For #3773

andreilupu committed:
Add gutenberg_content_has_block and gutenberg_post_has_block functionality on server side.

Also, added some phpunit tests for these functions.

For #3773

andreilupu committed:
fix phpcs warnings

andreilupu committed:
fix phpcs warnings

andreilupu committed:
Merge remote-tracking branch 'upstream/master' into 3773-add-has-block-functions

mzorz committed:
adds export getBlockTypes to be used in react native (#8457)

Ipstenu committed:
Fixing the inner block link (#8476)

Update link to inner blocks

Shelob9 committed:
Use @wordpress/compose instead of  @wordpress/components (#8537)

These docs show deprecated usage of withState

Rahmon committed:
Fix Custom HTML error when block is empty (#8556)

ajitbohra committed:
Fix link to server-side-render component (#8546)

bradyvercher committed:
Export the description for server-registered blocks. (#8555)

gziolo committed:
Docs: Update links to resources moved to packages folder (#8572)

jasmussen committed:
Polish focus and active styles, and do cleanups (#8385)

* Fix regression with button active style.

* Move from 1px to using $border-width

This is sort of some cleanup work across the files.

* Try new input focus style.

Fixes #7053.

youknowriad committed:
Data Module: Adding The queried data handling into the entities abstraction (#8357)

* Data Module: Adding The queried data handling into the entities abstraction

Co-authored-by: aduth <andrew.duthie@automattic.com>

* Fix a single entity fetch

* Data: Vary received entities action by query presence

* Data: Normalize queried pagination values as numeric

* Data: Support deep value serialization of query key

youknowriad committed:
DomReady: Fix race condition (#8578)

swissspidy committed:
Translatable demo content (#8551)

Translate Gutenberg demo content

Fixes #8547.

sarahmonster committed:
Fix awkward capitalisation in demo post content; fixes #8590. (#8591)

youknowriad committed:
API: Replace all withAPIData usage and deprecate the HoC (#8584)

* API: Replace all withAPIData usage and deprecate the HoC

* Replace withAPIData usage in docs

* Fix unit tests

youknowriad committed:
Inserter: Try making the inline tokens less important (#8587)

* Inserter: Try making the inline tokens less important

* Move the inline elements after the most used blocks

* Fix inline tokens test

aduth committed:
Docs: Add @mirucon to CONTRIBUTORS.md

Commit:
Reset previous changes, add CSS to do the job instead (#7751)

youknowriad committed:
Avoid using elt.remove because it's unavailable in ie11 (#8598)

aduth committed:
TinyMCE: Remove container assignment hack (#8465)

designsimply committed:
Add a note about handling blocks with `save` (#8154)

* Add a note about handling blocks with `save`

Fixes #6356. Adds documentation about how to handle blocks with `save` that don't support an `id` prop noted in #6356.

* Docs: Add note about handling attributes in custom save components

gziolo committed:
Updated dependencies: `jest`, `npm-package-json-lint` and `read-pkg-up` (#8596)

## Description

This PR updates `npm-package-json-lint` to the latest version and adds newly introduced rules. Props to @tclindner for making it possible:

> Hey! I just released v3.3.0 of npm-package-json-lint with a new description-format rule. It has two options:
> 
> requireCapitalFirstLetter - Throws an error if the first character in the description isn't capitalized.
> requireEndingPeriod - Throws an error if the description doesn't end with a period.
> Hopefully this works well for your team!

With this change, we can integrate new linting rule which will ensure that `description` inside `package.json` file will have proper formatting.

I took advantage of the fact that we are changing deps and bumped also `jest` which had also minor release.

I also downgraded `read-pkg-up` to avoid having it installed locally in the package. Instead, we can use the older version which is used with other packages. It doesn't break anything and keeps our setup simpler.

## How has this been tested?
`npm test` should cover all tools.

aduth committed:
DOM: Restore horizontal edge traversal implementation (#8461)

* Add a failing test for removing an empty first line.

* DOM: Restore horizontal edge traversal implementation

Co-Authored-By: iseulde

tomjn committed:
swaps out the CSS2 :before with the CSS3 ::before, see #8618 (#8630)

noisysocks committed:
Fix unrecognised `forwardedRef` prop (#8570)

* Fix iframe JavaScript indendation

* Fix unrecognised `forwardedRef` prop error

Have `withGlobalEvents` omit the `forwardedRef` prop. This prevents the
prop being added to HTML in some cases, e.g. when used in
`FocusableIframe`.

jasmussen committed:
Fix issue with link dialog. (#8646)

aduth committed:
Data: Add redux-routine package for synchronous generator flow (#8096)

* Packages: Add package `redux-routine`

* Packages: Throw rejected promise as error to generator

* Data: Add support for controls via redux-routine

* Core Data: Partially reimplement resolvers as controls

* Data: Port controls as data plugin

* Data: Revert deprecation of async generator

* Docs: Add note about plugin opt-in for controls

* ReduxRoutine: Improve isGenerator accuracy

* Fix redux-routine unit tests

nosolosw committed:
Allow plugins to extend the block settings menu (#7895)

fabianpimminger committed:
Let Wordpress make images in gallery block responsive / add srcset attribute  (#8593)

* Add wp-image-ID class to gallery to make images reponsive

* edit tests to include class name

youknowriad committed:
RichText: Removing/merging richtext should only trigger if the selection is collapsed (#8651)

* RichText: Removing/merging richtext should only trigger if the selection is collapsed

* Add e2e test to avoid uncollapsed selection to be mergedwq

* Code Style improvements per review

notnownikki committed:
Move embed API call out of block and into data (#6678)

aduth committed:
Redux Routine: Remove fake timers from tests (#8661)

earnjam committed:
Fix thead colspan bug (#7899)

danielbachhuber commented:
Related #8352

danielbachhuber commented on lib/blocks.php:
```diff
@@ -208,3 +208,55 @@ function do_blocks( $content ) {
+ * @param string $content Content to test. Fallback on the $post global's content in case it's missing.
+ * @return bool Whether the content contain the specified block.
+ */
+function gutenberg_content_has_block( $block_name, $content = null ) {
```
I don't love this magic behavior of `$content = null`. If I passed a `$content` variable that was set to `null`, then I'd unexpectedly get the global `$post->post_content`.

Is there prior art in core that you're basing this on?

andreilupu replied:
I was trying to be consistent with `gutenberg_post_has_block` but now I'm having second thoughts.
`gutenberg_post_has_block` is related to `$post` and prior in core, I can think of `has_excerpt`.

But in opposition, I think that `gutenberg_content_has_block` should be specific with params and no magic effects.

I'll make it required. Do you think that `gutenberg_post_has_block` should keep the `global $post` default?

danielbachhuber replied:
> Do you think that `gutenberg_post_has_block` should keep the `global $post` default?

Isn't `gutenberg_post_has_block` a new function? I think it should be passed an explicit string instead of any dependency on the `$post` object.

Taking a step back though, I think we should wait until further progress on #8352 until we spend more time on this. It may be that the parser itself exposes an equivalent method.

danielbachhuber commented:
Looks like the `gutenberg_` prefix might be dropping #8631

adamsilverstein committed:
Initialize the Autosaves controller, even if revisions are disabled (#8682)

notnownikki committed:
Move CI back to latest WordPress version and bump minimum version to 4.9.8 (#8454)

* Back to latest WordPress version

* Update WordPress is the requested version is 'latest'

* REST API: Remove polyfill for assign_terms visibility

* Framework: Bump WordPress compatibility to 4.9.8

danielbachhuber committed:
Remove REST API shims for code introduced in WP 4.9.8 (#8695)

* Remove REST API shims for code introduced in WP 4.9.8

Now that WP 4.9.8 is Gutenberg's minimum version, this is legacy code
that can be removed.

* Restore compatibility shim, as this hasn't landed in core yet

tofumatt committed:
chore: Add test for block mover (#8011) (#8392)

* chore: Add test for block mover (#8011)

wpscholar committed:
Add `gutenberg_can_edit_post` filter (#8323)

ntwb committed:
Update `package-lock.json` (#8700)

## Description

Fixes #8684

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

• Ensured my local Git repo was up to date with no changed files.
```
❯ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

• Removed my local `node_modules` folder:
```
❯ rm -i -rf node_modules/
```

• Ran `npm install`
```
❯ npm install
```
• Ran `git status`
```
❯ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   package-lock.json

no changes added to commit (use "git add" and/or "git commit -a")
```

• The `package-lock.json` diff is the same as @aduth notes in #8684 

• Ran `npm test`
```
Test Suites: 1 skipped, 244 passed, 244 of 245 total
Tests:       1 skipped, 2081 passed, 2082 total
Snapshots:   74 passed, 74 total
Time:        72.278s
Ran all test suites.
```


## Screenshots <!-- if applicable -->

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

Build Tools 

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

Commit:
Set babel dependencies to fixed version and added core-js2 support (#8586)

* Set babel dependencies to fixed version and added core-js2 support for babel-plugin-trnasform-runtime

* Packages: Start using @babel/runtime-corejs2 with the latest Babel

aduth committed:
Packages: Redux Routine: Add Babel runtime dependency (#8660)

* ReduxRoutine: Add Babel runtime dependency

* Redux Routine: Use @babel/runtime-corejs2 dependency

* chore: Update package-lock.json file

* tests: Update failing test by regenerating snapshot

noisysocks committed:
Add stylelint for SCSS linting (#8647)

* Add stylelint

Adds stylelint and configures it to lint SCSS files with a very
conservative rule set.

* Make "npm run lint" run its tasks concurrently

mmtr committed:
Add components examples in README.md files - Part 2 (#8338)

* Components examples: Modal

* Components examples: Modal

* Components examples assigned to const variables

* Components examples: Notice

* Components examples: Panel

* Components examples: Placeholder

* Components examples: Popover

* Components examples: QueryControls

* Components examples: RadioControl

* Components examples: RangeControl

* Components examples: ResponsiveWrapper

* Components examples: SandBox

* Components examples: ScrollLock

* Components examples: SelectControl

* Components examples: ServerSideRender

* Add metadata using YAML front matter so examples can be extracted
dynamically

* Add metadata using YAML front matter so examples can be extracted
dynamically

* Remove YAML front matter

* Components examples: SlotFill

* Components examples: Spinner

* Components examples: TabPanel

* Components examples: TextControl

* Components examples: TextareaControl

* Components examples: ToggleControl

* Components examples: Toolbar

* Components examples: Tooltip

* Components examples: TreeSelect

* Components examples: NavigableContainer

* Components examples: navigateRegions

* Components examples: withConstrainedTabbing

* Components examples: withContext

* Components examples: withFallbackStyles

* Components examples: withFilters

* Components examples: withFocusOutside

* Components examples: withFocusReturn

* Components examples: withNotices

* Components examples: withSpokenMessages

* Components examples: withAPIData

* Contributing guidelines for the components

* Contributing guidelines for the components

* Contributing guidelines for the components

* Add dummy attributes to ServerSideRender example

nosolosw committed:
Remove unused terms, taxonomies and categories code (#8250)

* Deprecate receiveTerms

* Deprecate terms and taxonomies reducers

* Deprecate getCategories resolver

* Deprecate getCategories and getTerms selectors

* Deprecate isRequestingTerms and isRequestingCategories selectors

* Update docs

* Do not deprecate reducers, as they are not public API

* Use public API in deprecation messages

* Update docs

* Resolve conflicts in deprecations file

* Use 3.7.0 version

* Teach tests to understand deprecation warnings

aduth committed:
Framework: Remove deprecations slated for 3.5 removal (#8687)

* Framework: Update use of deprecated UID

* Framework: Update use of deprecated compose

* Framework: Remove deprecations slated for v3.5 removal

* Framework: Update package dependencies after deprecation removals

aduth committed:
DOM: Abort selection range set on unset range target (#8691)

pento committed:
Add a warning when Cloudflare blocks REST API requests (#8640)

* Catch a Cloudflare response, and redirect to the Classic editor..

* Ensure the url package is loaded when api-fetch needs it.

* Show a message in the Classic Editor when we redirect there from a Cloudflare block.

* Don't auto-redirect to Classic, give a link, instead.

* Linting.

* Pinking shears

* Fix the incorrect Cloudflare rule ID being listed.

* Mock Resonse.clone() in the tests.

* Make checkCloudflareError() more resilient to non-strings being passed to it.

* Use typeof instead of checking for .indexOf on the error to check that it's a string.

* s/login/log in/

* Improve translator comments.

* Improve readability of the JSON parsing error handling

* Add a comment explaining why the response is being cloned.

nosolosw committed:
Add nosolosw as a Contributor (#8705)

As per:

https://github.com/WordPress/gutenberg/issues?utf8=%E2%9C%93&q=is%3Aissue+author%3Anosolosw+
https://github.com/WordPress/gutenberg/pulls?utf8=%E2%9C%93&q=author%3Anosolosw+is%3Apr

dimadin committed:
I18n: Consolidate similar strings (#8708)

gziolo committed:
Try to rewrite e2e tests using jest-puppeter preset (#8301)

* Try to rewrite e2e tests using jest-puppeter preset

* Testing: Unskip NUX tests

* Testing: Unskip 4 more e2e tests

* Testsing: Unskip all e2e tests

* Tests: Add one more check for undo e2e tests

* Testing: Improve preview e2e test

* Tests: Use explicit `setViewport` calls

* Build: Ensure `npm run build` is always executed before e2e tests

* Tests: Change the trigger wchich finishes after test cleanup

* Stop using `targetcreated`, see https://github.com/berstend/puppeteer-extra/issues/6

* lint fix

* Use `waitFor` instead of custom sleep

* Fix conflicts with master

* Lint fixes

* Update comment on race condition workaround

* Tests: Improve lifecycle methods for e2e setup

* Send more mousemove events, wait for selector before clicking

* chore: Improve variable name

* chore: Add colon to "See" comment

* chore: Tweak comments in undo test

* chore: Move puppeteer config to e2e folder

* chore: Introduce new e2e helper for clicking block appender

* Tests: Improve stability of e2e tests

* tests: Make e2e tests work with the lastest master changes

* tests: Add puppeteer config back

wpscholar committed:
Add "components" docs (#8337)

* Auto-generate components documentation manifest

* Ensure that the packages root doc is defined immediately before its children.

* Ran `npm run docs:build`

* Automated build of docs based on changes to the `root-manifest.json` file, any change to the docs tooling, and package files that would likely require rebuilding the docs. Also automated running of js and package.json linting. All occurs on a pre-commit hook.

* Ensure that `packages/*/README.md` file changes also trigger docs generation.

* Rename `Components` parent title to `Components Package Reference`

* Ensure that `Packages` shows first, then `Data Package Reference`, then `Components Package Reference`

* Automatically ran `npm run docs:build`

* Split up manifest.js into separate functions and removed code that generated manifest from config.js. Regenerated docs.

* Update package-lock.json

* Update data-core.md

* Update manifest.json

* Update data-core-editor.md

* Update manifest.json

noisysocks committed:
NUX: Prevent duplicate DotTips from appearing (#8642)

When there are multiple instances of `DotTip` that reference the same
tip identifier, only show the first instance that was mounted.

jasmussen committed:
Polish the checkbox, update switch documentation (#8588)

* Polish the checkbox, update switch documentation

The switch has a long history. Before I go into details, here's what this PR does:

- It replaces Pending Review, Stick to Front Page, Allow Comments, and Allow Pingbacks from using Switches to using Checkboxes.
- It tweaks the design of the checkbox to have a significantly more visible checked state, and a new more visual focus state.
- It tweaks the documentation for when to use a switch instead of a checkbox.

The Switch is born from mobile, which is increasingly the device of choice for people and long term might become the only choice. It's bigger and more tappable, and implies by its nature that it's a boolean choice. It also has qualities that the checkbox does not embody:

- Like its wall lightswitch counterpart, it’s something you flip and the effect is instant. There used to be darkness, now there’s light.
- It has only a single purpose, and that purpose is boolean. For example you wouldn’t use switches to select multiple categories.

However Allow Comments and the three other aforementioned toggles that use switches in master, do not have an instant effect, they require you to hit "Save" before the toggle option is stored. Given the above reasoning, a checkbox is a better choice in those situations.

The toggle is still a good choice in situations where immediate visual feedback is given, like when a dropcap is applied, or when you toggle image croppping for gallery items. In those situations, the pairing of the boolean nature and the larger touch target makes the switch feel more substantial.

With this PR I'm trying to take into account all the past discussions on the topic, and I know there have been many. Lay your thoughts on me.

* docs: Tidy up FormToggle docs

* Removed unused variables and withInstanceId.

Thanks for the feedback, @paulwilde.

* Try a fix for the legitimate failing e2e test.

Not sure if this is kosher, as the ID will change if additional checkboxes are added.

* Fix e2e test really.

aduth committed:
Blocks: Render Button RichText as div (#8451)

* Blocks: Render Button RichText as div

* Blocks: Update Button block snapshot per changed element

youknowriad committed:
onWriting Flow: Remove the provisional block behavior (#8706)

Commit:
Bug fixes for full/wide image alignments. (#8464)

* Show inspector controls on images that aren't resizable

This ensures inspector controls for alternate text and link settings are still available whenever an image is not resizable by including inspector controls always, and only conditionally loading specific controls that are relevant to sizing whenever isResizable is true.

* Reset image source size when using wide/full alignments.

This resets the `url` attribute of an image to the full size source URL whenever changing alignments to 'wide' or 'full', which prevents a blurry image whenever someone selects a smaller source image before changing to a 'wide' or 'full' alignment.

* Sanity check for the source image before resetting the url attribute

This fixes an issue where updating alignments on an image block that doesn't have a related attachment object results in an error.

* Allow choosing the image size in wide/full images as well

ianbelanger79 committed:
Expand date filter select box width (#8335)

* Expand date filter select box width

Widens the "All Dates" select box in the "Add Media" modal window, so that the full placeholder text is shown. Fix for #2202

* Fix indentation issues

sarahmonster committed:
Improve pre-publish panel styling (#7879)

* Use clearer wording for pre-publish panel prompt; fixes #7873.
* Style pre-publish panel more consistently with settings sidebar; fixes #7878.
* Switch to an all-white background for prepublish panel.

vindl committed:
Packages: update api-fetch package documentation (#8730)

Add a minor fix to the api-fetch middleware documentation
related to overriding the API root URL.

earnjam committed:
Prevent Travis from running when changes are only made to .md files (#8728)

This comes from [a discussion](https://wordpress.slack.com/archives/C02QB2JS7/p1533720310000209) with @gziolo in #core-editor. Basically we can skip the full Travis build if the commit range being tested only contains changes to `.md` files. 

Modeled after: https://github.com/facebook/react/pull/2000

In my testing, this shortens the Travis run time on each item in the test list from ~4-6 minutes down to ~1 minute or less for docs only changes. Should help speed up merging non-code PRs. 

Here are some test builds I did from my fork:
[Only change .md file](https://travis-ci.org/earnjam/gutenberg/builds/413732648) - aborts early
[Only change .js file](https://travis-ci.org/earnjam/gutenberg/builds/413733458) - runs normally
[Changes both .md and .js files](https://travis-ci.org/earnjam/gutenberg/builds/413733882) - runs normally

afercia committed:
Additional keyboaard shortcuts for navigateRegions. (#8005)

desrosj committed:
Manually print inline metabox initialization JavaScript when the `wp-edit-post` script is printed in the `<head>`.

Fixes #6963.

jasmussen committed:
Sort strikethrough after link (#8755)

This is a tiny PR to sort the "Strikethrough" button after the link.

One of the... perhaps this is unwritten and should be written down somewhere... one of the design principles of the block toolbar is that it starts with block related commands (like switcher), then _block level_ (i.e. actions that apply to the whole block, like textalign applies for the whole paragraph), and finally _inline level actions_ last. In the latter, it is sorted by presumed order of importance, and in this case, I would like to argue that Link is more important than Strikethrough.

One side effect of this is that one day when we can make the toolbar better handle nested contexts or small screens, we can "pop off" buttons into an overflow menu as the toolbar has to scale. If/when that happens, it's important that buttons are also sorted in order of importance.

noisysocks committed:
Fix 'Prevent duplicate DotTips from appearing' feedback (#8741)

* Typo: s/dispathed/dispatched/

* NUX: Simplify REGISTER_TIP_INSTANCE reducer

koke committed:
Expose redux-routine to React Native (#8761)

aduth committed:
Testing: Trigger E2E test failure on console logging (#8721)

* Testing: Extract post trashing to function

* Testing: Remove all page events on teardown

* Testing: Trigger E2E test failure on console logging

ntwb committed:
Update Lerna to 3.0.0-rc.0 (#8759)

ntwb committed:
chore(release): publish

 - @wordpress/a11y@1.1.3
 - @wordpress/api-fetch@1.1.1
 - @wordpress/autop@1.1.3
 - @wordpress/babel-plugin-import-jsx-pragma@1.0.3
 - @wordpress/babel-plugin-makepot@2.0.3
 - @wordpress/babel-preset-default@2.0.4
 - @wordpress/blob@1.0.3
 - @wordpress/blocks@2.0.0
 - @wordpress/components@2.0.0
 - @wordpress/compose@1.0.3
 - @wordpress/core-data@1.1.0
 - @wordpress/custom-templated-path-webpack-plugin@1.1.4
 - @wordpress/data@1.2.0
 - @wordpress/date@1.0.3
 - @wordpress/deprecated@1.0.3
 - @wordpress/dom-ready@1.2.0
 - @wordpress/dom@1.0.3
 - @wordpress/editor@2.0.0
 - @wordpress/element@1.0.3
 - @wordpress/hooks@1.3.3
 - @wordpress/html-entities@1.0.3
 - @wordpress/i18n@1.2.3
 - @wordpress/is-shallow-equal@1.1.3
 - @wordpress/jest-console@2.0.3
 - @wordpress/jest-preset-default@2.0.3
 - @wordpress/keycodes@1.0.3
 - @wordpress/library-export-default-webpack-plugin@1.0.3
 - @wordpress/nux@1.1.0
 - @wordpress/plugins@1.0.3
 - @wordpress/postcss-themes@1.0.3
 - @wordpress/redux-routine@1.0.0
 - @wordpress/scripts@2.1.0
 - @wordpress/shortcode@1.0.3
 - @wordpress/url@1.2.3
 - @wordpress/viewport@1.0.3
 - @wordpress/wordcount@1.1.3

diegoreymendez committed:
Publishes unknown-type handler methods for mobile. (#8688)

dimadin committed:
Capital P, dangit (#8772)

gziolo committed:
Element: Improve validation for attribute names in serializer (#8775)

aduth committed:
RichText: Fix empty value delete behaviors (#8735)

* Testing: Wait for RichText initialization

* RichText: Fix empty value delete behaviors

* RichText: Handle remove only on backspace

* docs: Very minor comment readability tweak

mcsf committed:
i18n: Add context to occurrences of 'More' string (#8766)

aduth committed:
Update plugin version to 3.5 (#8784)

Commit:
Localization & Spelling fixes (#8795)

Fixes #8794.

talldan committed:
Add new keyboard shortcuts for block settings menu (#8279)

Adds new keyboard shortcuts for the block settings menu:

Cmd+Shift+D / Ctrl+Shift+D - duplicate block
Cmd+shift+, / Ctrl+shift+, - show/hide sidebar

* Add duplicate shortcut to block settings menu

- Move onDuplicate dispatch from BlockDuplicateButton into a new HOC called
`withBlockSettingsActions`
- Add `shortcuts` as a prop in `withBlockSettingsActions` that is passed to
its wrapped component
- Pass onDuplicate and shortcut through to BlockDuplicateButton
- Add new BlockSettingsKeyboardShortcuts component for handling shortcut
functionality

Make BlockDuplicateButton a dumb component - move HOC logic to withBlockSettingsActions

Add keyboard shortcut for block removal (Cmd/Special/Windows + DEL)

- Move block remove action dispatch from BlockRemoveButton up to withBlockSettingsActions
- Make BlockRemoveButton a dumb component
- Add keyboard shortcut for block removal to BlockSettingsKeyboardShortcuts
- Pass onRemove into KeyboardShortcuts and BlockRemoveButton

Add shortcut for toggling the sidebar

- Add shortcut keys to list of keyboard shortcuts for edit-post
- Add open/close dispatchers to EditorModeKeyboardShortcuts and hook up
- Make BlockInspectorButton display the keyboard shortcut

MenuItem should use aria-checked attribute instead of aria-pressed

See example implementation of a MenuBar here for details:
https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-2/menubar-2.html

Provide a default role for MenuItem component of menuitem

Still allows the role to be overriden for other menuitem types
(i.e. menuitemcheckbox, menuitemradio), but defaults to menuitem.

Also makes the provision of isSelected prop optional since the resulting
aria-checked role should not be used with a menu-item.

Update usages of MenuItem that should not use the default role

Switch menu items in block settings menu that have a shortcut to use MenuItem

Use the MenuItem component which correctly renders the shortcut. The need for
wrapping components BlockDuplicateButton and BlockRemoveButton is minimal, so
remove them.

Remove role props, since MenuItem implements a default

Make all block settings menu items use MenuItem component instead of IconButton

Spread additional props for MenuItem onto rendered component

This allows MenuItem to support additional props like `disabled`

Remove stroke being applied to MenuItem icons

Most icons don't have stroke - this style rule was adding an additional stroke
to SVG icons, making them appear chunky and blurry.

Remove use of !important

Fix lack of left margin on menu items with icons

Fix potential access of undefined property

When removing the block, there's momentarily a null item in the blocks array

Render block settings when typing, but make it hidden

This ensures BlockSettingsKeyboardShortcuts is rendered and shortcuts
work while actively typing

* Rename shortcut keycodes to raw/display to match block settings menu

* displayShortcut removes `+` symbol between shortcuts when the provided key is any single character

* displayShortcut capitalizes provided key (e.g. Del instead of DEL)

* Avoid placing a + between ⌘ and key entirely

* Change duplicate shortcut to Cmd/Ctrl shift D

* Change toggleSidebar shortcut to `cmd/ctrl + shift + ,` to avoid conflict with Chrome settings shortcut

* When toggling the sidebar using the keyboard, open the correct tab based on whether block(s) are selected

* Add keyboard shortcuts to tooltips for closing and opening the settings sidebar

* Make use of stopImmediatePropagation less zealous

Code is supposed to prevent further handling of keyboard events when a block is
deleted, but doesn't consider cases where a block is not deleted:

- If the user hits delete and the block is kept, but content from another block
merged in
- If the user hits delete, but the block has content and is thus not deleted

* Add modifier key exceptions to prevent deletion or merge of blocks

* Make block deletion shortcut Cmd+Opt+Backspace or Cmd+Opt+Delete

- For Windows that's Ctrl+Alt+Backspace or Ctrl+Alt+Delete

* Add further clauses where event should return early

Iron out inconsistent behaviour with use of backspace/delete key when it can
result in deletion or merging of blocks. When specific modifiers are pressed
the event should not have any effect. Here, the use of preventDefault and
stopImmediatePropagation were blocking further binding of keyboard shortcuts
and preventing browser/system level keyboard shortcuts.

* Make keyboard shortcut for delete Cmd+Alt+Delete

* Abbreviate display shortcut to prevent line wrapping in block settings menu

* Merge `withBlockSettingsActions` into `BlockSettingsMenu`

- BlockSettingsMenu was the only consumer of withBlockSettings
- shortcuts object is now not injected into props

* Remove BlockSettingsKeyboardShortcuts component, which had no logic

* Use preventDefault to prevent any additional browser/os shortcuts and add comments

* Use object method shorthand

* Add !important back in, with clarifying comment

* Use lodash noop over noop arrow function

* Destructure dispatching functions

* Refine exceptions to RichText Backspace/Delete handling

* Remove the Remove Block shortcut

* Revert changes to RichText back to master version

* Fix unit tests (stylelint)

talldan committed:
Add new keyboard shortcuts help modal (#8316)

Pressing ctl+option+h (shift+alt+h on windows/linux) displays a modal detailing keyboard shortcuts.

* Create a simple modal for showing keyboard shortcuts that displays when pressing cmd + /

* Add more shortcuts and place them in sections with a title and description

* Add styles for shortcut modal

* Add missing shortctuts

* Remove unecessary shortcut section descriptions

* Display shortcut keys in a kbd element

* Styling changes

- Position shortcut to the right of description
- Adjust margin/padding
- Make section titles default font size

* Regroup shortcuts

* Fix invalid style rule

* Change copy for slash inserter shortcut

* Map over config to produce ShortcutSections

* Change where keyboard shortcut help modal is rendered

Move modal so that it's rendered in edit-post/layout component. This ensures
the modal is available in both the text and visual editor, and the shortcut
for launching the modal can be used on first load of the editor.

* Move keyboard-shortcut-help-modal to edit-post from editor package

* Add some border between rows in the keyboard shortcuts list

* Add reduxy things for opening and closing a modal

* Use actions to open/close modal and selector to determine current state

* Migrate the activeModal state from preferences to normal state so that its value is not persisted

* Add a menu item tot he more menu for toggling keyboard shortcut help

* Modify some shortcuts and description text

* Add snapshot tests for help modal

* Destructure props

* Modify header styling

* Rename class name to reflect component is now in edit-post, and reorder shortcuts

* Display shortcut for more menu menu item

* Disable WP_Help TinyMCE command for classic block

* Change keyboard shortcut for triggering help to access + h, the same as the classic editor

* Add utility to keycodes for returning key combinations in an array

* Avoid ugly regular expression to split key combination into array

Using newly introduced displayShortcutList function results in cleaner, better tested code

* Add e2e test for shortcut help modal

* Vertically center shortcut text (if adjacent label wraps onto multiple lines)

* Close more menu when keyboard shortcuts modal is opened using the menu item

* Use castArray over lengthy ternary

* Styling linter fixes

* JSDoc fixes

* Group activeModal unit tests in a describe

* Add test for initial activeModal state

* Fix incorrect state used in test

* Update list of key shortcuts for the help modal

* Remove unecessary return

* Make KeyboardShortcutsHelpMenuItem only open the help modal

jasmussen committed:
Pad menus (#8758)

* Pad menus

This adds padding to the block more menu. At first glance this will look like much, but in fact it simply makes it the same padding as that of every other popover.

* Sort again.

koke committed:
Port heading block to mobile (#8774)

* Adds tagName prop to RichText

Instead of forcing a paragraph, make the wrapper tag logic more generic
and allow for customization

* Adds React Native implementation of heading block

* Fix lint issues

* Remove RichText tagName default value

* Remove unused (for now) class name props

youknowriad committed:
Audio Block: Avoid scrollbars when shown full-width (#8820)

jasmussen committed:
Make active pill button look pressed. (#8819)

* Make active pill button look pressed.

This is a small change that makes the activated pill button look pressed. Right now it just looks like a button, and while it is, it's an _active_ button.

* Fix typo.

notnownikki committed:
Allow access to the WordPress installation if DOCKER_ENV=localwpdev (#8818)

youknowriad committed:
Extract Core Blocks to the npm packages folder (#8287)

* Extract Core Blocks to the npm packages folder

* Core blocks: Adjustment after rebase with master

* Build: Add support for all scss files in build-style directory

* Build: Fix output path for styles generated for packages

* Fix sass files after rebase

* Fix watching SASS files

* Rename core-blocks block-library

* Fix linting issues

* Update docs

* Deprecate Core Blocks

* Update index.js

* Update client-assets.php

* Update client-assets.php

chrisvanpatten committed:
Updates docs for templatelock's `insert` option (#8843)

danielbachhuber committed:
Remove TinyMCE shim what was shimmed in WP 4.9.7 (#8832)

aduth committed:
Framework: Remove client-side document title updates (#8831)

* Framework: Deprecate DocumentTitle

* Edit Post: Remove DocumentTitle from editor layout

* Always force `<title>` to 'Edit Post' (or equivalent)

It needs to be in a more generic state. Setting the global `$title`
overrides the behavior of `get_admin_page_title()` while still letting
the value be filterable via `admin_title`.

* Use `/*` for inline comments

* Catch deprecations in tests

* Remove tests around deprecated functions

* Fix PHPCS issue

johngodley committed:
Fix deletion of custom block classes (#8232)

* Fix deletion of custom block classes

If you add a custom class to a block and then remove that custom class it would flag the block as invalid. The reason is that the custom class became part of the blocks ‘default’ attributes, and it would then add it back and this then the expected HTML wouldn’t match the current HTML.

This change looks at the difference between the default class and the inner HTML.

The unit tests have been beefed up to cope with several edge case situations

danielbachhuber committed:
Ensure meta box functions are available in Gutenberg context (#8850)

jasmussen committed:
Update Dashicons component. (#8811)

* Update Dashicons component.

This adds a few new icons, notably Insert Before and Insert After.

But it also adds any dashicons that have been added to the upstream repo in the mean time.

* Update to match gutenberg style.

This is also accompanied by an upstream PR to the Dashicons repo.

pento committed:
Add a package license checker (#8808)

Ensure that all packages we distribute have a GPL2 compatible license, and any other packages we use during development have some form of OSS license.

Fixes #6508, #7822.

youknowriad committed:
Fix unhandled error dialog styling (#8834)

mmtr committed:
Add styles for Spinner component (#8801)

* Add styles for Spinner component

* Add styles for Spinner component

* Improvements to the Spinner component.

- Rotation animation can be reused
- Background color set from variable
- Fix tests

* Fixing stylelint issue

* Fixing stylelint issue

* Trigger Travis

* Trigger Travis

roborourke committed:
URLInput: Pass the the search result object to props.onChange (#8662)

* Pass the the search result object to props.onChange

This component is great however the number of use cases it can support increases a huge amount if it makes the search result object available to the `onChange()` prop as well.

Example use cases:

- Retrieving the post title, ID and post type, as well as URL to update block attributes
- Fetching other / additional post data using the `_links` property of the search result object

* Add a component readme

* Fix json syntax

* Fix code example indentation

* document URLInputButton as well

* No autoFocus prop for button

* Remove unneeded arg on onChange method

Thanks for the review @youknowriad

* Update docs to reflect nullable `post` arg [ci skip]

* remove first arg from selectLink

* Improve readability

Add some changes based on feedback from @tofumatt, much appreciated!

youknowriad committed:
IE11: Fix default appender placeholder (#8837)

gziolo committed:
chore(release): publish

 - @wordpress/block-library@1.0.0
 - @wordpress/blocks@2.0.1
 - @wordpress/components@2.0.1
 - @wordpress/compose@1.0.4
 - @wordpress/core-data@1.1.1
 - @wordpress/data@1.2.1
 - @wordpress/editor@2.0.1
 - @wordpress/element@1.0.4
 - @wordpress/keycodes@1.1.0
 - @wordpress/nux@1.1.1
 - @wordpress/plugins@1.0.4
 - @wordpress/scripts@2.2.0
 - @wordpress/viewport@1.0.4

youknowriad committed:
Data Module: Correctly handle per_page=-1 in the queried data state (#8908)

youknowriad committed:
Test: Ensure the full content integration test is run (#8923)

dsawardekar committed:
Adds support for matching blocks with diacritics using lodash.deburr (#8707)

* Adds support for matching blocks with diacritics using lodash.deburr.

* Updates inline doc for normalizeTerm

* Updates normalizeTerm comment message for clarity

* Inserter: JSDoc improved style standards

Period after sentence. Newline before return statement. 80 column line-width.

nosolosw committed:
PrePublishPanel: suggest tags and postformat (#8235)

dimadin committed:
i18n: Use argument swapping instead of named arguments for string placeholders (#8771)

* Use argument swapping instead of named arguments for string placeholders

* Use singular or plural string based on number of selected blocks.

Commit:
[Fix] Make sure Latest Posts alignment class behaviour (#8847)

* Make sure Latest Posts block doesn't apply alignment classes if no alignment is selected

* Blocks: Fix Latest Posts align assignment alignment

notnownikki committed:
Remove the workaround for intermittent multiple-tab preview test failure (#8928)

SergioEstevao committed:
Fix paragraph splits on RN (#8915)

* Make sure regex is applied in multiple lines.

* Add entry point for React Native.

talldan committed:
Add remove block keyboard shortcut (#8805)

Adds the keyboard shortcut Ctrl+Alt+Backspace / Alt+Cmd+Backspace for removing a block

* Add shortcut for Remove Block

* Add shortcut for remove block to shortcut help modal

* Allow event for remove block to bubble from RichText when the shortcut chord is pressed

* Use `isKeyboardEvent` to catch key combination + improve comment

hypest committed:
Add the React Native entrypoint to all runtime packages (#8933)

* Add entry point for React Native.

* Add the React Native entrypoint to more packages

So that the packages that are needed in the runtime (like, not in the
build phase) define it for the React Native mobile app to properly
resolve them.

sarahmonster committed:
Update & flesh out design handbook (#8924)

* Adding revised design handbook strucutre + content

* Removing old Design doc from the Reference section.

* Properly closing the code block in the theme example.

* Properly opening the code block in the theme example.

* Rename pages to prevent the repeated use of "Design"

"Design Patterns" > "Patterns"
"Design Resources" > "Resources"

These all live under the "Design" folder anyway, so I think it's fine to remove that part of the title.

* Add new pages/structure to root-manifest.json

* Update filenames + slugs.

Changing "patterns" to "design-patterns", and "resources" to "design-resources" to prevent conflict with other slugs we're already using.

* Wee copyedits to main design page.

* Minor copyedits to block design page.

* Copyedit design resources page.

* Tiny copyedit for clarity on the design patterns page.

* Copyedits to the Design Principles & Vision page

props @michelleweber.

* More copyedits.

props @michelleweber

gziolo committed:
Components: Deprecate `withContext` HOC and remove its usage (#8189)

* Components: Deprecate withContext HOC

* Element: Add support for Context API in serializer

* Element: Make serializer work with nested context consumers and providers

* Element: Improve detection for Context API components

* Components: Update deprecation version for wp.component.withContext

* Blocks: Simplify BlockContent context HOC implementation

* Blocks: Addressing feedback from review

* Element: Add test for serialization with nested providers

* Test: Fix test description in serializer test suite

* Tests: Mock deprecated when testing if it was called

nosolosw committed:
Inserter: take into account nested blocks to render the list of insertable blocks (#8075)

Previous to this, it was possible to add single blocks (those which have `multiple: false`) more than once if a post/page by using nested blocks.

youknowriad committed:
Fix the shortcode block UI in IE11 (#8922)

youknowriad committed:
Tags: Fix the tags input in IE11 (#8918)

* Tags: Fix the tags input in IE11

* Min-height instead of line-height for inputs

Commit:
[Enhancement] Change color of links in gallery block caption (#8004)

* Change color of links in gallery block caption

* Last minute changes

* chore: Tweak RGBA styles in SCSS

youknowriad committed:
Fix our custom element serializer in IE11 (#8967)

chrisvanpatten committed:
Apply new checkbox CSS to radio buttons and fix border radius (#8835)

* Apply new checkbox CSS to radio buttons w/ border-radius fix; fixes #8833

* Use $round-radius instead of 50%

talldan committed:
Add new Insert Before/After menu items and shortcuts (#8909)

Adds the following keyboard shortcuts for those actions:

Option+Cmd+T (Mac) / Ctrl+Alt+T - Insert Before
Option+Cmd+Y (Mac) / Ctrl+Alt+Y - Insert After

Changes:
* Add basic insert before and insert after buttons to block settings menu

* Add keyboard shortcuts for insert before and insert after

* Change insert before/after shortcuts to meta+alt+t and meta+alt+y

* Add comment explaining use of preventDefault

* Add insert before/after shortcuts to keyboard help

* Do not show Insert Before/After menu items when a block is template locked

* Disallow block settings actions that can be triggered through shortcuts and should not be possible when template locking is active

* Tidy up comments

  - Add full-stops
  - Remove comment that no longer seems relevant (component is not a DropDown)

prajapatisagar committed:
Remove unnecessary comma ( , ) (#8910)

jasmussen committed:
Fix code block style regression. (#9011)

The recent input field changes regressed the style of the code block. This fixes them again, and adjusts the padding to be less rectangular.

chrisvanpatten committed:
Refactor template select field to use SelectControl (#8987)

* Refactor the template select field to use SelectControl; fixes #8985

* Simplify onChange prop

aduth committed:
Preview: Open new window if prior preview window closed (#8446)

aduth committed:
Default Block Appender: Remove redundant event handlers (#8714)

aduth committed:
Element: Create dummy context components for type switch (#8998)

youknowriad committed:
Fix canvas jumb when navigation regions (#9014)

youknowriad committed:
Bring the preview tab to the front when clicking the preview button (#9015)

aduth commented:
Is this the same as #8631 ?

jsnajdr committed:
toAsyncIterable: remove Promise.resolve call that's already done by the JS runtime (#8913)

When `await`-ing something or `yield`-ing something from an async iterator, one doesn't
need to call `Promise.resolve` to cast it to a Promise. The language runtime and all
library functions already to that for us.

andreilupu commented:
@aduth I don't think they are the same.

What I'm proposing in this PR is a way of checking if a post or a piece of content has a specific `block_type`. 
For example, prior in core, we had the `has_shortcode` function, and we could easily identify if a page has a `gallery` shortcode so can enqueue a weird js slider functionality only for that page.

As for 8631, I think that it adds a check if there is any kind of block type in content in a more generic way, which could be useful to add a body or post class if the looped post is edited with Gutenberg.

Both PRs are necessary for plugin and theme developers IMO

aduth commented:
Okay, thanks for the clarification. I assume then it _is_ the same function, `has_blocks`, with overloaded arguments to optionally specify the specific block type for which to check the presence?

andreilupu commented:
Technically, it may be, and I actually use that function in my PR to optimize the check here https://github.com/WordPress/gutenberg/pull/8340/files#diff-1334f2cb73e38f2f89d2aaf763f07ed9R258

I tend to slightly disagree with unifying these functions because we lose consistency with the ol' core functions like `has_shortcode`.
Also I'm not finding a good sematic in `has_plural( $of_singular )` but I won't hold to this because I'm not a native English speaker.

I don't have a personal attachment to the PR; I'm happy to follow a decision as long as we have to functionality available in some form.

Side question: I added some PHPUnit tests here, are they useful? Or I was overthinking at the time?
In case they are good, I could help with more tests in other PRs / functionalities.

obenland commented:
Similar to `WP_Hook::has_filters()` and `WP_Hook::has_filter()`, I think `has_block( $block, $post )` would be a good complement to `has_blocks()`. I'd also advocate for merging `gutenberg_post_has_block()` and `gutenberg_content_has_block()` and just have `has_block()` be able to handle both.

stoyan0v committed:
Show drop zone background when file is dragged (#8752)

* Fixes regression preventing background color from showing when dropping a file onto a block

aduth commented:
>I tend to slightly disagree with unifying these functions because we lose consistency with the ol' core functions like has_shortcode.

This assumes there is consistency 😄 See also [`has_excerpt`](https://developer.wordpress.org/reference/functions/has_excerpt/).

aduth commented:
So pluralization is good to raise. The function here is `has_block` (single, presence of one of a given type by argument), while the function in #8631 is `has_blocks` (plural, presence of any, no arguments). To me, this seems like a sensible separation.

aduth commented:
The tests look well written to me 👍

Commit:
Fix small typo (#9020)

aduth committed:
Edit Post: Reset active sidebar tab on initial load (#8995)

* Edit Post: Reset active sidebar tab on initial load

* Edit Post: Set default active general sidebar in reducer

* Testing: Add E2E test case for sidebar dismissed preference

obenland committed:
Add generic has_blocks() function (#8631)

This deprecates `gutenberg_post_has_blocks()`.

See #4418, #8352.

johngodley committed:
Fix crashed block height and inserted order (#8926)

* Crash block: fix min height

Needs to be twice as tall otherwise it will overlap other blocks

* Crash block: allow inserter to appear over block

youknowriad committed:
Publish button: Avoid changing the label of the "publish" button if an auto-save is running (#9024)

jasmussen committed:
Polish permalink UI and make it responsive (#9010)

* Fix input and focus style for permalink slug

* Fix regression with horizontal scrollbar on mobile.

* Use serious flexing to scale the permalink editor.

* chore: Fix typo

* chore: Wording tweak

* Clarify comments.

* Address feedback.

* chore: Fix failing CSS Lint test

jasmussen committed:
Release font-size for code. (#8914)

* Release font-size for code.

This addresses feedback in this comment https://github.com/WordPress/gutenberg/pull/710/files/3096d0bebe5d64142a84511b954d89ea6599e4f3#r208286450

Previously, the font size for any element in the `code` tag was fixed to 14px. Now it's released, free, and can be any size the parent container suggests to it.

Note that the `inherit` is necessary here to override code from upstream.

* Add comment to note the reason for inherit.

jasmussen committed:
Let the menu grow (#8756)

* Let the menu grow

This PR lets the menu grow to accommodate long text. Already now in Spanish, the "Code Editor" wraps having to fit both the spanish label and keyboard shortcut. As we add more keyboard shortcuts as well as hints, we need to let the menu grow to accommodate that, up to the mobile breakpoint in width.

This PR accomplishes that using auto widths, nowrap whitespace, and a little Nacin' Spacin' thrown in for code cleanup while I was there.

* Add case for mobile

* Try and clarify comment.

mmtr committed:
Add styles for Notice component (#8856)

coderkevin committed:
Data: Add RegistryConsumer export (#9042)

The plugin code relies on the RegistryProvider to be able to add
functionality to the registry. `withSelect` uses the RegistryConsumer,
but code outside of `@wordpress/data` can't access this modified
registry at all (even though it can create it!).
This solves the incongruence by simply exporting RegistryConsumer as
well.

tofumatt committed:
feat: Hide keyboard shortcuts on mobile screens (#9051)

* feat: Hide keyboard shortcuts on mobile screens

Fix #9050.

* docs: Clarify reasoning for wanting more space

* Add min-width to menu.

This fixes a regression I introduced in https://github.com/WordPress/gutenberg/pull/8756/files#diff-dcb8eefee4aaf82c9ddb042494547527R158.

youknowriad committed:
IE11: Fix The metaboxes area position (overlapping the content) (#8964)

* IE11: Fix The metaboxes area position (overlapping the content)

* Remove unnecessary CSS rule

youknowriad committed:
Fix the block styles in IE11 (#8962)

* Fix the block styles in IE11

* Remove extra padding

youknowriad committed:
Fix the heading level buttons in IE11 (#8840)

* Fix the heading level buttons in IE11

* Fix mobile toolbars

* Fix heading buttons in IE11

tofumatt committed:
feat: Show sidebar in HTML edit mode for blocks (#8969)

* feat: Show sidebar in HTML edit mode for blocks (fix #4413)

* docs: Update inspector docs

* docs: Add docs to component

* docs: Improve image in inspector docs

* chore: Improve conditional and fix unintended isSelected prop

* docs: Change docs from master

aduth committed:
Framework: Remove deprecations slated for 3.6 removal (#8993)

* Shared Blocks: Add deprecation notices for deprecated shared blocks

* Framework: Remove deprecations slated for 3.6 removal

mtias committed:
Custom Block Icons (#8916)

* Custom icons for Blocks

Uses custom svgs at a 24px base, 36px upscale for the inserter.
WordPress should signal that block icons ought to be custom for the
sake of clarity for users.

* Update some more icons.

* Add icons for several core blocks.

* Use service icons for embeds.

* Adjust embed icons.

* Expand width of inserter, use lighter design.

* Use ligher line arrow for panel collapse / expand.

* Adjust inserter design giving more paddings.

* Update more block icons and use color API for embeds.

* Adjust svg sizes for non-dashicon icons.

* Adjust block icon stack.

* Use different icons for embed fallback.

* Add more embed logos (WordPress, Tumblr).

* Add remaining icons for code, shortcode, inline image.

* Simplify image SVG and add new Heading icon.

* Polish inline image & block chip

This changes the icon for inline images to be bigger, and consistent with the image block.

It also polishes the block chip to ensure it has the same size in the inspector and in the toolbar, and have the right size.

* Cleanup SVG declarations.

* Fix appender icons

* Normalize SVGs and fix slash command

* Normalize focus style in library.

* Fix test issues.

* Clean "Classic" icon and linter issues.

* Polish focus, hover styles.

* Tweak how scrolling area works, and add grid units.

* It's padding, not margin. Sigh.

* Fix switcher

* Fix style variations.

* Fix unit tests

* Refactoring block icons rendering

* Fix style lint

* Remove debug code :) Sorry

* Fix icon styles

* Fix block icons e2e test

* Make sure the e2e icons test validates the colors properly

obenland committed:
Introduce strip_dynamic_blocks() for excerpts  (#8984)

Initially used when generating excerpts to avoid infinite loops when dynamic blocks generate excerpts, and to add parity with `strip_shortcodes()`.

Fixes #5572, #7268.

andreilupu committed:
Remove prefix and make it consistent with has_blocks

andreilupu committed:
merge

andreilupu commented:
My "rebase" went horribly wrong and I can't manage to clean it properly.

In lack of time I recreated the PR here #9058; I hope it's ok

pento commented:
Thanks @andreilupu. Just for reference, it's okay if the you accidentally include a bunch of commits while rebasing, we squash PRs when they're merged, so it doesn't have an effect on what's actually committed.

To do a clean rebase of a PR, you can run this command: `git pull --rebase origin master`
